src/main/java/com/revature/controllers/CurriculumModuleController.java

```diff
@@ -0,0 +1,58 @@
+package com.revature.controllers;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.revature.entities.CurriculumModule;
+import com.revature.services.CurriculumModuleService;
+
+@CrossOrigin(origins = "*", allowCredentials="true")
+@RestController
+@RequestMapping("/curriculum-modules")
+public class CurriculumModuleController {
+
+	@Autowired
+	CurriculumModuleService currModuleService;
+	
+	@PostMapping
+	public ResponseEntity<CurriculumModule> createCurrModule(@RequestBody CurriculumModule cMod){
+		return ResponseEntity.ok(currModuleService.createCurrModule(cMod));
+	}
+	
+	@GetMapping
+	public ResponseEntity <Set<CurriculumModule>> getAllCurrModules(){
+		Set<CurriculumModule> currMods = currModuleService.getAllCurrModules();
+		return ResponseEntity.ok(currMods);
+	}
+	
+	@GetMapping(value="/{id}")
+	public ResponseEntity<CurriculumModule> getCurrModuleById(@PathVariable int id){
+		CurriculumModule currMod = currModuleService.getCurrModuleById(id);
+		return ResponseEntity.ok(currMod);
+	}
+	
+	@PutMapping("/update")
```

Comment:
y

Reply:
I don't know what you want me to do with this information

Reply:
"update" for a post request does not follow restful conventions! Just mentioned in CurriculumController comments. https://restfulapi.net can clarify naming conventions if they're still unclear